ecommers/lib/core/repositories/product_data_repository.dart

```diff
@@ -0,0 +1,137 @@
+import 'package:ecommers/core/common/index.dart';
+import 'package:ecommers/core/models/data_models/index.dart';
+import 'package:ecommers/core/models/index.dart';
+import 'package:ecommers/core/models/sort_type.dart';
+import 'package:ecommers/core/services/index.dart';
+import 'package:enum_to_string/enum_to_string.dart';
+
+class ProductDataRepository {
+  Future<List<Product>> getProductListByCategory({
+    String category,
+    String subCategory,
+    int rangeFrom,
+    int rangeTo,
+    String searchQuery,
+    SortType sortType,
+  }) async {
+    final cachedData = await _provideProductsByCategoryFromCache(category);
+    if (cachedData != null &&
+        cachedData.products != null &&
+        cachedData.products.isNotEmpty) {
+      if (DateTime.now().difference(cachedData.lastUpdatedDate).inDays >= 1) {
+        await cacheDatabase.deleteDataByFilter(
+            CacheDefines.products, 'key', category);
```

Comment:
what this 'key' means?

Reply:
This is the property by which we are trying to extract data. So probably, i can define it's as a const value.